internal/controller/storage/storage.go

```diff
@@ -0,0 +1,99 @@
+package storage
```

Comment:
This entire file can move to the `resources` package I think, no need for its own package